culinary-canvas/capstone-project/backend/index.js

```diff
@@ -57,6 +88,187 @@ const storage = new CloudinaryStorage({
 });
 const upload = multer({ storage: storage })
 
+//find the lowest Nutrient
+const findLowestNutrient = (nutrientTotals) => {
+    let lowestNutrient = null;
+    let lowestValue = Infinity;
+
+    Object.keys(nutrientTotals).forEach((key) => {
+        if(nutrientTotals[key] < lowestValue){
+            lowestNutrient = key;
+            lowestValue = nutrientTotals[key]
+        }
+    });
+    return lowestNutrient;
+};
+
+const capitalizeFirstLetter = (string) => {
+    return string.charAt(0).toUpperCase() + string.slice(1).toLowerCase()
+}
+
+//fetch recipes high in the lowest nutrient
+const fetchRecipesHighInNutrient = async(nutrient) => {
+    try{
+        const response = await fetch(`https://api.edamam.com/search?q=${nutrient}&app_id=${EDAMAM_APP_ID}&app_key=${EDAMAM_APP_KEY}`);
+        const data = await response.json();
+        if (!data.hits){
+            throw new Error('No recipes found!')
+        }
+        const capitalizedNutrientLabel = capitalizeFirstLetter(nutrient)
+        const filteredRecipes = data.hits.filter(hit => {
+            const totalNutrients = hit.recipe.totalNutrients;
+            for (const nutrient in totalNutrients){
+                if (totalNutrients[nutrient].label === capitalizedNutrientLabel){
+                    return true;
+                }
+            }
+            return false;
+
+        })
+        return filteredRecipes.map(hit => ({
+            label: hit.recipe.label,
+            url: hit.recipe.url,
+            image: hit.recipe.image,
+            calories: hit.recipe.calories
+        }));
+
+
+    }catch(error){
+        console.error(`Error fetching recipes high in ${nutrient}: `, error)
+        return [];
+    }
+
+};
+
+//fetch user notifications
+const fetchUserNotifications = async(userId) => {
+    try{
+        const userPreferences = await prisma.userPreferences.findUnique({
+            where: {userId : userId}
+        });
+
+        const mealPlanner = await prisma.mealPlanner.findMany({
+            where: {userId : userId}
+        });
+
+        const recipeIds = mealPlanner.map(entry => entry.recipeId)
+
+        const fetchRecipeData = async(id)=>{
+            try{
+                const response = await fetch(`https://api.edamam.com/search?r=${encodeURIComponent(id)}&app_id=${EDAMAM_APP_ID}&app_key=${EDAMAM_APP_KEY}`);
```

Comment:
is fetch a function you wrote somewhere? Or is this the whole API call, was just surprised there's not as much post-processing as I'd expect.

Reply:
This is the API call